homeassistant/components/mqtt/client.py

```diff
@@ -618,6 +621,9 @@ def async_remove() -> None:
             """Remove subscription."""
             self._async_untrack_subscription(subscription)
             self._matching_subscriptions.cache_clear()
+            # Make sure we allow retained payloads when resubscribing
```

Comment:
I think this comment is misleading? The dictionary keys are the id of `Subscription` objects, so even if we don't do the cleanup here we'll get retained messages if we do exactly the same subscribe again?